File: docs/js/edge.js
/**
 * Ребра графа
 * @param {int} square Ход
 * @param {Vertex} vertex Следующая вершин
 */
function Edge(square, vertex) {
    this.square = square;
    this.vertex = vertex;
}

Edge.prototype.compareTo = function (e) {
    return this.vertex.compareTo(e.vertex);
};


Edge.Comparator = function (e1, e2) {
    if (e1 === null) {
        return e2 === null ? 0 : 1;
    }
    return e1.compareTo(e2);
};

File: src/java/org/webjs/rv2/Edge.java
package org.webjs.rv2;

/**
 * Ребра графа
 *
 * @author rmr
 */
public class Edge implements Comparable {

    public Edge(int square, Vertex vertex) {
        this.square = square;
        this.vertex = vertex;
    }

    // Ход
    public int square;

    // Следующая вершина
    public Vertex vertex;

    @Override
    public int compareTo(Object o) {
        return vertex.compareTo(((Edge) o).vertex);
    }

}
